connect/src/spirc.rs

```diff
@@ -976,11 +977,19 @@ impl SpircTask {
                 return self.notify().await;
             }
             Play(play) => {
-                let context_uri = play
-                    .context
-                    .uri
-                    .clone()
-                    .ok_or(SpircError::NoUri("context"))?;
+                let context = match play.context.uri {
+                    Some(ref s) => PlayContext::Uri(s.clone()),
```

Comment:
Always gets me :-) a reference to a cloned value, can we do better?

Reply:
So I tried a bit around to get a cleaner solution but it's hard to do. 

For context: I have to do this here, because I move the context later, and if it moved partially before it's not possible to move the entire context. I will continue trying to improve the handling but it's a very tricky spot

Reply:
No biggie... probably have better things to spend your time on :smile:

Reply:
I like solving those kind of problems, so I will see what I can do :). But might take some time as motivation and time is a bit limited currently.

Reply:
So this should do the trick code wise. But it will require some testing before being merged as I adjusted some other parts which might impact other cases.

Reply:
So I tested the spots where I suspected some side effect, but it seems fine and should hopefully have no side effects.

> The changes I was concerned about where in `merge_context`